src/main/java/org/launchcode/water_garden_tour/controllers/GardenController.java

```diff
@@ -63,6 +65,99 @@ public String listGarden(Model model) {
         return "gardens/list";
     }
 
+    @GetMapping("/gardens/admin-list")
+    public String listAdminGarden(Model model) {
+        model.addAttribute("gardens", gardenRepository.findAll());
+        model.addAttribute("title", "Garden ADMIN List");
+
+        return "gardens/admin-list";
+    }
+
+    @PostMapping("/gardens/delete")
+    public String deleteGarden(Model model, @RequestParam int gardenId) {
+
+        Optional optGarden = gardenRepository.findById(gardenId);
+        if (optGarden.isPresent()) {
+            Garden garden = (Garden) optGarden.get();
+
+            gardenRepository.delete(garden);
+        }
+
+        model.addAttribute("gardens", gardenRepository.findAll());
+        model.addAttribute("title", "Garden ADMIN List");
+
+        return "gardens/admin-list";
+    }
+
+    @GetMapping("/gardens/update")
```

Comment:
@jamiander did you say you were routing to /gardens/update as well? could your work be combined with what I have here? I can change my routing if that's best!

Reply:
This is looking really good!  I like the Admin page!  My code, after gardens/add redirects to gardens/uploadImage (for convenience) where admin can upload or chose no upload.  It then redirects to gardens/list.  Gardens/uploadImage can also be done separately from adding a garden if admin needs to add something later, so it might be a good idea to put a link to it on the Admin page.

Reply:
Sounds good! And we can always add some flair to the admin page later if we have time.